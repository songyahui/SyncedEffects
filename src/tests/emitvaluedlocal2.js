"use hopscript"

var hh = require("hiphop");

function sum(arg1, arg2) {
   return arg1 + arg2;
}

var prg = <hh.module O>
  <hh.loop>
    <hh.local S=${{initValue: 1}}>
      <hh.emit S apply=${function() {return this.S.preval + 1}}/>
      <hh.emit O apply=${function() {return this.S.nowval}}/>
    </hh.local>
    <hh.pause/>
  </hh.loop>
</hh.module>

exports.prg = new hh.ReactiveMachine(prg, "emitvaluedlocal2");
