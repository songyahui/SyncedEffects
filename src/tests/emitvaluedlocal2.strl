module emitvaluedlocal2:

output O: integer;

loop
	signal S := 1 : combine integer with + in
	       emit S(pre(?S) + 1);
	       emit O(?S);
	end signal;
	pause
end loop

end module
