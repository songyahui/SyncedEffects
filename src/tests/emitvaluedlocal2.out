emitvaluedlocal2> ;
--- Output: O(2)
emitvaluedlocal2> ;
--- Output: O(2)
emitvaluedlocal2> ;
--- Output: O(2)
